handlers/handlers.go

```diff
@@ -46,74 +55,284 @@ func parseReqBody(w http.ResponseWriter, body io.ReadCloser, bodyObj *models.Pre
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(prefs)
 }
 
 // DeletePrefsHandler deletes a user's preferences
-func DeletePrefsHandler(w http.ResponseWriter, r *http.Request) {
-	queryValues, err := url.ParseQuery(r.URL.RawQuery)
+func (env *Env) DeletePrefsHandler(w http.ResponseWriter, r *http.Request) {
+	vals, err := parseStringToInt(r.Header.Get("User-ID"))
 	if err != nil {
-		errMsg := "Failed to parse query: " + err.Error()
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if err = env.DB.DeletePrefs(vals[0]); err != nil {
+		errMsg := fmt.Sprintf(
+			"unable to delete preferences for user: %s",
+			err.Error(),
+		)
+		responseCode := http.StatusInternalServerError
+		if err == models.ErrPrefsDNE {
+			errMsg = err.Error()
+			responseCode = http.StatusNotFound
+		}
 		log.Println(errMsg)
-		http.Error(w, errMsg, http.StatusBadRequest)
+		http.Error(w, errMsg, responseCode)
+		return
+	}
+
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// DeletePrefsConvHandler deletes a user's preferences for a conversation
+func (env *Env) DeletePrefsConvHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	vals, err := parseStringToInt(r.Header.Get("User-ID"), vars["conversation"])
+	if err != nil {
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
-	resBody := map[string]string{
-		"user_id":         queryValues.Get("user_id"),
-		"conversation_id": queryValues.Get("conversation_id"),
+	if err = env.DB.DeletePrefsConv(vals[0], vals[1]); err != nil {
+		errMsg := fmt.Sprintf(
+			"unable to delete preferences for user: %s",
+			err.Error(),
+		)
+		responseCode := http.StatusInternalServerError
+		if err == models.ErrPrefsDNE {
+			errMsg = err.Error()
+			responseCode = http.StatusNotFound
+		}
+		log.Println(errMsg)
+		http.Error(w, errMsg, responseCode)
+		return
 	}
 
-	json.NewEncoder(w).Encode(resBody)
+	w.WriteHeader(http.StatusNoContent)
 }
 
 // PatchPrefsHandler updates a user's preferences
-func PatchPrefsHandler(w http.ResponseWriter, r *http.Request) {
+func (env *Env) PatchPrefsHandler(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close()
 
-	reqBody := models.NewPreferences()
+	reqBody := &models.GlobalPrefs{}
+	if err := parseReqBody(w, r.Body, reqBody); err != nil {
+		return
+	}
+
+	vals, err := parseStringToInt(r.Header.Get("User-ID"))
+	if err != nil {
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if err := env.DB.PatchPrefs(vals[0], reqBody); err != nil {
+		errMsg := fmt.Sprintf(
+			"unable to update preferences for user: %s",
+			err.Error(),
+		)
+		responseCode := http.StatusInternalServerError
+		if err == mongo.ErrNoDocuments {
+			errMsg = err.Error()
+			responseCode = http.StatusNotFound
+		}
+		log.Println(errMsg)
+		http.Error(w, errMsg, responseCode)
+		return
+	}
+
+	json.NewEncoder(w).Encode(reqBody)
```

Comment:
Does this need to set the `Content-Type` header to `application/json` like the other handlers with response bodies?

Reply:
Oh lol I forgot this one I guess